File: 62.unique-paths.py
class Solution:
    def uniquePaths(self, m, n):
        """
        :type m: int
        :type n: int
        :rtype: int
        """
        co = [0] * (m + n - 1)
        co[0] = 1
        for i in range(1, m + n - 1):
            for j in range(i, 0, -1):
                co[j] = co[j] + co[j - 1]
        return co[m - 1]

File: 62.unique-paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        vector<int> co(m + n - 1, 0);
        co[0] = 1;
        for (int i = 1; i < m + n - 1; i++) {
            for (int j = i; j > 0; j--) {
                co[j] = co[j] + co[j - 1];
            }
        }
        return co[m - 1];
    }
};